optihood/energy_network.py

```diff
@@ -246,7 +297,7 @@ def createNodesData(self, data, filePath, numBuildings, clusterSize):
             nodesData["electricity_cost"].index = pd.to_datetime(nodesData["electricity_cost"].index, format='%d.%m.%Y %H:%M')
 
         if "naturalGasResource" in nodesData["commodity_sources"]["label"].values:
-            if type(natGasImpact) == float or (natGasImpact.split('.')[0].replace('-','').isdigit() and natGasImpact.split('.')[1].replace('-','').isdigit()):
+            if isinstance(natGasImpact, (float, np.float64)) or (natGasImpact.split('.')[0].replace('-','').isdigit() and natGasImpact.split('.')[1].replace('-','').isdigit()):
```

Comment:
This relates to the different ways in which the parser and readers treat the data.